src/util/__tests__/createServerRootMixin.test.js

```diff
@@ -519,63 +514,76 @@ Array [
 
       await renderToString(wrapper);
     });
-  });
-
-  describe('hydrate', () => {
-    it('sets __initialSearchResults', () => {
-      const serialized = createSerializedState();
 
+    it('searches only once', async () => {
+      const searchClient = createFakeClient();
       const app = {
         mixins: [
+          forceIsServerMixin,
           createServerRootMixin({
-            searchClient: createFakeClient(),
+            searchClient,
             indexName: 'hello',
           }),
         ],
-        render(h) {
-          return h(InstantSearchSsr, {}, [
+        render: h =>
+          /**
+           * This code triggers this warning in Vue 3:
+           * > Non-function value encountered for default slot. Prefer function slots for better performance.
+           *
+           * To fix it, replace the third argument
+           * > [h(...), h(...)]
+           * with
+           * > { default: () => [h(...), h(...)] }
+           *
+           * but it's not important (and not compatible in vue2), we're leaving it as-is.
+           */
+          h(InstantSearchSsr, {}, [
             h(Configure, {
               attrs: {
                 hitsPerPage: 100,
               },
             }),
             h(SearchBox),
-          ]);
-        },
-        // in test, beforeCreated doesn't have $data yet, but IRL it does
-        created() {
-          this.instantsearch.hydrate({
-            __identifier: 'stringified',
-            hello: serialized,
-          });
+          ]),
+        serverPrefetch() {
+          return this.instantsearch.findResultsState(this);
         },
       };
 
-      const {
-        vm: { instantsearch },
-      } = mount(app);
+      const wrapper = new Vue({
+        mixins: [forceIsServerMixin],
+        render: h => h(app),
+      });
 
-      expect(instantsearch.__initialSearchResults).toEqual(
-        expect.objectContaining({ hello: expect.any(SearchResults) })
-      );
+      await renderToString(wrapper);
 
-      expect(instantsearch.__initialSearchResults.hello).toEqual(
-        expect.objectContaining(serialized)
-      );
+      expect(searchClient.search).toHaveBeenCalledTimes(1);
+      expect(searchClient.search.mock.calls[0][0]).toMatchInlineSnapshot(`
+Array [
+  Object {
+    "indexName": "hello",
+    "params": Object {
+      "facets": Array [],
+      "hitsPerPage": 100,
+      "query": "",
+      "tagFilters": "",
+    },
+  },
+]
+`);
     });
+  });
 
-    it('accepts non-stringified results', () => {
+  describe('hydrate', () => {
+    it('sets _initialResults', () => {
```

Comment:
I see there was a conversation to discard this test [here](https://github.com/algolia/vue-instantsearch/pull/1090/files?show-viewed-files=true&file-filters%5B%5D=#r764181800), but it wasn't removed in the end. Any reason not to? Anything that it tests that isn't covered elsewhere?

Reply:
It's indeed testing some InstantSearch internals, but the main reason it stayed is that Vue InstantSearch is extending InstantSearch with a couple methods, which could in theory cause hydrate not to work correctly anymore